runbotics/runbotics-orchestrator-ui/src/src-app/views/credentials/GridView/CredentialsGridView.tsx

```diff
@@ -0,0 +1,151 @@
+    const collections = useSelector(state => state.credentialCollections.credentialCollections);
+    const credentialTemplates = useSelector(state => state.credentialTemplates.data);
+    const [loading, setLoading] = useState(true);
+
+    const credentialsPage = useSelector(state => state.credentials.page);
+    const pageFromUrl = searchParams.get('page');
+    const [page, setPage] = useState(pageFromUrl ? parseInt(pageFromUrl, 10) : 0);
+    const pageSizeFromUrl = searchParams.get('pageSize');
+    const [pageSize, setPageSize] = useState(pageSizeFromUrl ? parseInt(pageSizeFromUrl, 10) : 12);
+    const totalPages = Math.ceil(credentials.length / pageSize);
+    const startingPageItemIndex = page * pageSize;
+    const endingPageItemIndex = startingPageItemIndex + pageSize;
+    const currentPageCredentials = credentials ? credentials.slice(startingPageItemIndex, endingPageItemIndex) : [];
+
+    const [currentDialogCredential, setCurrentDialogCredential] = useState(null);
+    const [isDeleteDialogOpen, setIsDeleteDialogOpen] = useState(false);
+
+    const router = useRouter();
+    const collectionId = router.query.collectionId ? (router.query.collectionId as string) : null;
+
+    const handleDeleteDialogOpen = (id: string) => {
+        setIsDeleteDialogOpen(true);
+        setCurrentDialogCredential(credentials.find(credential => credential.id === id));
+    };
+
+    const handleDeleteDialogClose = () => {
+        setIsDeleteDialogOpen(false);
+        setCurrentDialogCredential(null);
+    };
+
+    const handleDelete = () => {
+        dispatch(credentialsActions.deleteCredential({ resourceId: currentDialogCredential.id }))
+            .then(() => {
+                enqueueSnackbar(translate('Credential.Tile.Delete.Success', { name: currentDialogCredential.name }), {
+                    variant: 'success'
+                });
+                handleDeleteDialogClose();
+            })
+            .catch(error => {
+                enqueueSnackbar(error ? error.message : translate('Credential.Tile.Delete.Fail', { name: currentDialogCredential.name }), {
+                    variant: 'error'
+                });
+            });
+    };
+
+    useEffect(() => {
+        const fetchData = async () => {
+            try {
+                await dispatch(credentialCollectionsActions.fetchAllCredentialCollections());
+                await dispatch(credentialTemplatesActions.fetchAllTemplates());
+                collectionId
+                    ? await dispatch(credentialsActions.fetchAllCredentialsInCollection({ resourceId: `${collectionId}/credentials/` }))
+                    : await dispatch(credentialsActions.fetchAllCredentialsAccessibleInTenant());
+            } finally {
+                setLoading(false);
+            }
+        };
+
+        fetchData();
+    }, []);
+
+    useEffect(() => {
+        const pageNotAvailable = credentialsPage && page >= totalPages;
```

Comment:
again, how can this happen? if there is some code setting the page over the limit, it should be fixed there